CoreVisualGraph/src/au/gov/asd/tac/constellation/graph/visual/framework/GraphVisualAccess.java

```diff
@@ -1026,24 +1034,68 @@ public ConstellationColor getBlazeColor(final int vertex) {
         return blaze == null ? VisualGraphDefaults.DEFAULT_BLAZE_COLOR : blaze.getColor();
     }
 
+    private String getDecoratorStr(final int decoratorAttrib, final int decoratorVertex) {
+        // If a valid attribute ID has been supplied, return its value as a string. This will be mapped to an icon
+        // for display as a decorator. Boolean attributes are special cases as it is possible to overload the icon
+        // used for true and false values based on attribute name.
+        // This is acheived in the following way:
+        //   * The value string returned for boolean attributes is appended with the text .<attribute_name> to result
+        //     in a value of the form <value>.<attribute_name>
+        //   * In a class overriding ConstellationIcon ensure an icon is added with a name, or an alias matching 
+        //     the string <value>.<attribute_name>
+        // The icon DefaultIconProvider.TRANSPARENT is able to be aliased to support NOT showing an icon. Ie to NOT
+        // show an icon when a pinned attribute is set to false, add an alias to DefaultIconProvider.TRANSPARENT
+        // with name "false.pinned"
+        if (decoratorAttrib != Graph.NOT_FOUND)
+        {
+            final String value = accessGraph.getStringValue(decoratorAttrib, accessGraph.getVertex(decoratorVertex));
+            // If the attribute is a boolean, construct a value string including the attribute name as well,
+            // as per comments above.
+            if (accessGraph.getAttributeType(decoratorAttrib) == BOOLEAN_ATTRIBUTE_STR)
+            {   
+                // Booleans will either have a value of true.<attributeName> or false.<attributeName>
+                // there are three cases to cater for: 
+                // 1. Both true.<attributeName> and false.<attributeName> are set as aliases for icons
+                //    --> In this case we use the supplied icon
+                // 2. Only one of true.<attributeName> or false.<attributeName> is set as an alias for an icon
+                //    --> In this case the value tht doesnt have an icon set should display nothing
+                // 3. Neither true.<attributeName> or false.<attributeName> is set as ana alias
+                //    --> In this case there is no override, use the default true/false icons
+                final String notValue = this.bool_dict.get(value);
+                final String attributeName = accessGraph.getAttributeName(decoratorAttrib);
+                final String valueStr = value.concat(".").concat(attributeName);
```

Comment:
@serpens24 more just food for thought which is looking at the icon filename we have, the words are separated by the underscore instead of the period so was wondering if that would make more sense?

Reply:
I'll do this as it keeps it consistant.
.
I had toyed with using different dividers than _ as often _ is used to seperate wors.. so we could have had stuff like (if we had a pos_locked attribute (instead of pinned) ... true_pos_locked.png, whereas my apporach was true.pos_locked.png, but in reality it works either way.